Client/Frontend/Browser/TabManager.swift

```diff
@@ -315,7 +315,13 @@ class TabManager: NSObject {
         let type: TabType = isPrivate ? .private : .regular
         let tab = Tab(configuration: configuration, type: type)
         if !isPrivate {
-            tab.id = id ?? TabMO.create().syncUUID
+            if id != nil {
+                tab.id = id
+            } else {
+                DataController.performBackgroundTask { context in
```

Comment:
Not sure I follow this change. This function previously just used its own context and returned. Since this context isn't persisted (just used to pass into `create`), I think this is basically functionally equivalent, no?

Reply:
You are right, we could simplify this code and do `performBackgroundTask` inside of the `TabMO.create()` method.
Basically the only difference is that creating TabMO object is put on an operation queue

Reply:
I suggest not using any DataController functions directly in non NSManagedObject class(Unless required of course). Everything can be managed in the data classes themselves.